### PR TITLE
Bugfix - Typos

### PR DESCRIPTION
**Pull request by jutaz**
This PR fixes quite a few (if I do say so myself) typos found across docs and comments in this repo.

All were hand-picked, so should be good to go

### CODE REVIEW

**Comment by jutaz** [comment]
Signed, thanks for the note!

**Comment by runcom** [comment]
awesomeeee

File: registry/session.go [hunk 1]
```diff
@@ -100,8 +100,8 @@ func (tr *authTransport) RoundTrip(orig *http.Request) (*http.Response, error) {
 	// Authorization should not be set on 302 redirect for untrusted locations.
 	// This logic mirrors the behavior in addRequiredHeadersToRedirectedRequests.
 	// As the authorization logic is currently implemented in RoundTrip,
-	// a 302 redirect is detected by looking at the Referer header as go http package adds said header.
-	// This is safe as Docker doesn't set Referer in other scenarios.
+	// a 302 redirect is detected by looking at the Referrer header as go http package adds said header.
+	// This is safe as Docker doesn't set Referrer in other scenarios.
```
Comment (thaJeztah) [inline registry/session.go]: Hm, this may actually need to stay misspelled. For background, see https://en.m.wikipedia.org/wiki/HTTP_referrer

**Comment by thaJeztah** [comment]
This is really great, @jutaz :heart:

I left one note, but I'm also good leaving that in.

LGTM

**Comment by thaJeztah** [comment]
@runcom feel free to LGTM if it LGTY. I checked, and there's no vendored files in this PR, given the huge amount of files touched here, perhaps get it merged soon to prevent rebase hell

**Comment by runcom** [comment]
I was slowly checking :) 

LGTM

**Comment by thaJeztah** [comment]
Thanks @runcom!

(Janky is still running)

File: registry/session.go (hunk 1 above)
Comment (jutaz) [inline registry/session.go]: Hmm, I am unsure. This could stay in code as `referer`, while it would be referenced correctly in docs

Comment (thaJeztah) [inline registry/session.go]: Yes, I was in doubt as well, given that it is with a capital, I reckon it's referring to the name of the header. But I'm fine either way :smile:

**Pull request merged**